src/components/Modals/OnchainIdentityModal.js

```diff
@@ -84,24 +90,23 @@ function OnchainIdentityModal({
         name="email"
         placeholder="Email"
       />
-
-      <div className={styles.title}>Are you or do you want to become a citizen?</div>
-      <CheckboxInput
+      <div className={styles.title}>I am or want to become a</div>
+      <SelectInput
         register={register}
-        name="citizen"
-        setValue={setValue}
-        watch={watch}
-        label="I am or want to become a citizen"
+        name="onChainIdentity"
+        options={[
+          { value: "0", display: "E-resident", selected: !isCitizen && isEResident},
+          { value: "1", display: "Citizen", selected: isCitizen && isEResident},
+          { value: "2", display: "Neither", selected: !isCitizen && !isEResident},
```

Comment:
Can we have more descriptive `value`s here? Like `value: eresident`, `value: citizen` etc.?

Reply:
Yeah we can have strings here and in the code in general